File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.cpp
class Solution { // class Solution :
 public: // public:
  ListNode* swapPairs(ListNode* head) { // ListNodes* swapPairs(ListNode* head)
    const int length = getLength(head); // const int legth = getLength(head);
    ListNode dummy(0, head);  // ListNode dummy(0, head);
    ListNode* prev = &dummy;  //  ListNode* prev + &dummy;
    ListNode* curr = head; // ListNode* curr = head;

    for (int i = 0; i < length / 2; ++i) {  // for (int i = 0; i < lenggth / 2; ++i)
      ListNode* next = curr->next; //ListNode* next = curr->next;
      curr->next = next->next; //curr->next = next->next;
      next->next = prev->next; // next->next = prev->next;
      prev->next = next; // prev->next = next;
      prev = curr; //prev = curr;
      curr = curr->next; //curr = curr->next;
    }

    return dummy.next; //return dummy.next;
  }

 private: // private:
  int getLength(ListNode* head) { // int getLength(ListNode* head)
    int length = 0; // int length = 0;
    for (ListNode* curr = head; curr; curr = curr->next)  // for (ListNode* curr = head; curr; curr; curr = curr->next)
      ++length;
    return length; //return length;
  }
};

File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.java
class Solution {
        public ListNode swapPairs(ListNode head) {
                ListNode dummy = new ListNode(0);
                dummy.next = head;
                ListNode cur = dummy;
                while(cur != null && cur.next != null && cur.next.next != null){
                        ListNode temp = cur.next;
                        ListNode next = temp.next.next;
                        cur.next = temp.next;
                        cur.next.next = temp;
                        temp.next = next;
                        cur = cur.next.next;
                }
                return dummy.next;
    	}
}














